src/supautils.c

```diff
@@ -414,14 +413,13 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 							strcmp(defel->defname, "adminmembers") == 0)
 							hasrolemembers = true;
 
-						// Defer setting bypassrls attribute if using `privileged_role`.
-						//
-						// Duplicate/conflicting attributes will be caught by
-						// the standard process utility hook, so we can assume
-						// there's at most one bypassrls DefElem.
-						if (bypassrls_is_allowed && strcmp(defel->defname, "bypassrls") == 0) {
-							stmt_has_bypassrls = intVal(defel->arg) != 0;
-							intVal(defel->arg) = 0;
+						// Setting the superuser attribute is not allowed.
+						if (strcmp(defel->defname, "superuser") == 0) {
+							ereport(ERROR,
+									(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+									 errmsg("permission denied to alter role"),
+									 errdetail("Only roles with the %s attribute may alter roles with the %s attribute.",
+											   "SUPERUSER", "SUPERUSER")));
```

Comment:
@soedirgo Is this new? Would we need a new test for it? Assuming it's for `ALTER ROLE x SUPERUSER`.

Reply:
Yup, will add a test for it